desktop/cmp/grid/impl/filter/headerfilter/HeaderFilter.ts

```diff
@@ -31,32 +32,59 @@ export const headerFilter = hoistCmp.factory({
             onDoubleClick: stopPropagation,
             headerItems: [switcher()],
             item: tabContainer(),
-            bbar: bbar()
+            bbar: bbar(),
+            hotkeys: [
+                {
+                    allowInInput: true,
+                    combo: 'enter',
+                    label: 'OK',
+                    group: 'Column Filter',
+                    onKeyDown: () =>
+                        // Wait for debounced reaction in `ValuesTabModel` to run before committing
+                        wait(400).then(
+                            () => (model.hasFilter || model.hasPendingFilter) && model.commit()
+                        )
+                },
+                {
+                    allowInInput: true,
+                    combo: 'escape',
+                    label: 'Cancel',
+                    group: 'Column Filter',
+                    onKeyDown: () => model.parent.close()
+                }
+            ]
         });
     }
 });
 
 const bbar = hoistCmp.factory<HeaderFilterModel>({
     render({model}) {
-        const {commitOnChange} = model;
+        const {commitOnChange, hasFilter, hasPendingFilter, isDirty} = model;
         return toolbar({
             compact: true,
             items: [
-                filler(),
                 button({
                     icon: Icon.delete(),
-                    text: 'Clear Filter',
+                    text: 'Clear',
                     intent: 'danger',
                     disabled: !model.hasFilter,
                     onClick: () => model.clear()
                 }),
+                filler(),
                 button({
                     omit: commitOnChange,
-                    icon: Icon.check(),
-                    text: 'Apply Filter',
-                    intent: 'success',
-                    disabled: !model.hasFilter && !model.hasPendingFilter,
+                    text: 'OK',
+                    disabled: !hasFilter && !hasPendingFilter,
+                    intent: isDirty ? 'primary' : null,
+                    minimal: !isDirty,
+                    outlined: !isDirty,
                     onClick: () => model.commit()
+                }),
+                button({
+                    omit: commitOnChange,
+                    text: 'Cancel',
+                    outlined: true,
+                    onClick: () => model.parent.close()
```

Comment:
We put the OK/save/apply button to the right of cancel quite consistently across Hoist - would like to keep that ordering here.

Reply:
Would also drop outline on cancel - looks a bit weird in compact toolbar, clear filter not outlined